small-talk/app/chat/page.js

```diff
@@ -5,159 +5,128 @@ import socket from "../../util/socket";
 import React, { useEffect, useState } from 'react';
 
 export default function Chat(){
-	const [text, setText] = useState(null);
 	const [user, setUser] = useState([]);
 	const [selectedUser, setSelectedUser] = useState(null);
 	const [privateMessages, setPrivateMessages] = useState([]);
 	const [inputMessage, setInputMessage] = useState("");
 
 	useEffect(() => {
 		socket.auth = "random";
-    	socket.connect();
-		setText("whatever");
-
-		// getting all currently connected users
+		socket.connect();
+	
 		socket.on("users", (receivedUsers) => {
-            console.log("Received users:", receivedUsers);
-            receivedUsers.forEach((user) => {
-                user.self = user.userID === socket.id;
-            });
-            setUser(receivedUsers);
-        });
-
-		// set the event to connected users
+			receivedUsers.forEach((user) => {
+				user.self = user.userID === socket.id;
+			});
+			setUser(receivedUsers);
+		});
+	
 		socket.on("user connected", (user) => {
-			console.log("User connected:", user);
 			setUser(prevUsers => [...prevUsers, user]);
 		});
-
-		// deals with new messages from the recipient
+	
 		socket.on("private message", ({ content, from }) => {
-			console.log("got the private message yay: ", { content, from });
-			
-			setPrivateMessages(privateMessages => [...privateMessages, { content, from }]);
+			setPrivateMessages(prevMessages => [...prevMessages, { content, from }]);
+			console.log("private message:", privateMessages);
 		});
-
-		// Commented out because an aspect is missing and will be added in a future PR
-		// handles connection error
-		// socket.on("connect_error", (err) => {
-		// 	if (err.message === "invalid username") {
-		// 		this.usernameAlreadySelected = false;
-		// 	}
-		// });
-		
-		// will disconnect the socket
+	
 		socket.on("disconnect", () => {
 			console.log("User disconnected:");
-
-			setUser(prevUsers => {
-				return prevUsers.map(user => {
-					// if (user.self) {
-					// 	return{...user, connecteded: false};
-					// }
-					return user;
-				});
-			});
+			setUser(prevUsers => prevUsers.filter(user => user.userID !== socket.id));
 		});
-        
-		// disconnect sockets
+		
 		return () => {
 			socket.off("users");
 			socket.off("user connected");
-			//socket.off("connect_error");
 			socket.off("private message");
 			socket.off("disconnect");
 		};
-	}, [text]);
+	}, [privateMessages]); 
```

Comment:
Do we need this? I would've expected the private message socket to be able to handle changes to the chat box already

Reply:
I don't think that would change anything. I'll remove it.